sklearn/datasets/_kddcup99.py

```diff
@@ -45,7 +47,19 @@
 
 logger = logging.getLogger(__name__)
 
-
+@validate_params( 
+     { 
+         "subset": [StrOptions({"SA", "SF", "http","smtp"}),None],
+         "data_home":[str, None],
+         "shuffle":["boolean"],
+         "random_state":["random_state"],
+         "percent10":["boolean"],
+         "download_if_missing":["boolean"],
+         "return_X_y":["boolean"],
+         "as_frame":["boolean"],
+         
+     }     
+)
```

Comment:
This is still not black-complient. You can check my previous suggestion or just apply `black` on this file.